csrc/preseg_passes/reorder_sharded_axis.cpp

```diff
@@ -25,21 +25,16 @@ void ReorderShardedAxisPass::runPass(Fusion* fusion) {
   const std::vector<Expr*>& exprs = fusion->exprs();
   for (auto it = std::rbegin(exprs); it != std::rend(exprs); it++) {
     Expr* expr = *it;
-    if (!isResharding(expr)) {
+    if (HostIrLower::canLower(expr)) {
+      continue;
```

Comment:
Will this incorrectly skip non-contiguous allgather, e.g., `[i{M},iN]=Set([i{M},iDIDx{N}])`? AFAICS, [canLower](https://github.com/NVIDIA/Fuser/blob/e057a9bac33023a2a7c4ed2cc87ebe6fe3944d8f/csrc/host_ir/lower.cpp#L332-L333) doesn't check axis order for Set or Reduction.

Reply:
> AFAICS, [canLower](https://github.com/NVIDIA/Fuser/blob/e057a9bac33023a2a7c4ed2cc87ebe6fe3944d8f/csrc/host_ir/lower.cpp#L332-L333) doesn't check axis order for Set or Reduction.

Now it does, with the current patch!

> Will this incorrectly skip non-contiguous allgather, e.g., [i{M},iN]=Set([i{M},iDIDx{N}])?

So this will not skip a case like this one